### PR TITLE
Implement hide/show behavior on scroll in Blog Appbar

### PR DESCRIPTION
**Pull request by subbuwu**
This PR adds a sticky app bar to the project with hide/show behavior on scroll. The app bar remains visible when scrolling up and hides when scrolling down.

Video Attached : 
https://github.com/code100x/daily-code/assets/101008478/9ac18bb1-51f0-4807-9a33-9578ddea1061

### CODE REVIEW

File: packages/ui/src/BlogAppbar.tsx [hunk 1]
```diff
@@ -22,16 +22,34 @@ export const BlogAppbar = ({ problem, track }: { problem: Problem; track: Track
...
+    setScrollingDown(prevScrollPos < currentScrollPos);
+    setPrevScrollPos(currentScrollPos);
+  };
+
+  useEffect(() => {
+    window.addEventListener("scroll", handleScroll);
+    return () => {
+      window.removeEventListener("scroll", handleScroll);
+    };
+  }, [prevScrollPos]);
```
Comment (hkirat) [inline packages/ui/src/BlogAppbar.tsx:41]: Can you debounce the call to `handleScroll` ?

Comment (subbuwu) [reply packages/ui/src/BlogAppbar.tsx]: I have added debounce to handleScroll , you can review it and change the delay in ms as neeed .

Comment (subbuwu) [reply packages/ui/src/BlogAppbar.tsx]: https://github.com/code100x/daily-code/assets/101008478/e351766e-4a0a-4249-b50b-47c891bef7dd

**Pull request merged**